### ISSUE TITLE
Issues with Dynamic-Hooks RC-1

### ISSUE THREAD

**Issue by Bretto** [orientechnologies/orientdb#3250]
following the docs on Class level hooks: http://www.orientechnologies.com/docs/2.0/orientdb.wiki/Dynamic-Hooks.html
```
CREATE CLASS Invoice EXTENDS OTriggered
ALTER CLASS Invoice CUSTOM onAfterCreate=invoiceCreated
CREATE FUNCTION invoiceCreated "print('\\nInvoice created: ' + doc.field('number'));" LANGUAGE Javascript
INSERT INTO Invoice CONTENT { number: 100, notes: 'This is a test' }
```
error
```
orientdb {db=myDB}> INSERT INTO Invoice CONTENT { number: 100, notes: 'This is a test' }

Error: com.orientechnologies.orient.core.command.script.OCommandScriptException: Error on parsing script at position #0: Error on execution of the script
Script: invoiceCreated
------^

Error: javax.script.ScriptException: sun.org.mozilla.javascript.internal.EcmaError: ReferenceError: "doc" is not defined. (<Unknown source>#2) in <Unknown source> at line number 2

Error: sun.org.mozilla.javascript.internal.EcmaError: ReferenceError: "doc" is not defined. (<Unknown source>#2)
```

**Comment by nin-jin** [comment]
http://grepcode.com/file_/repo1.maven.org/maven2/com.orientechnologies/orientdb-core/2.0-rc1/com/orientechnologies/orient/core/db/record/OClassTrigger.java/?v=diff&id2=1.7.10#300

Fix it asap, please.

**Comment by orangemug** [comment]
Is there any target release for this bug? I need to implement audit tables and this doesn't seem possible while this is broken.

**Comment by lvca** [comment]
@nin-jin Sorry, with that link I don't understand what you're saying.

**Comment by orangemug** [comment]
Also is there any chance getting this removed from the docs while its broken? It was quite confusing as a new user to orientdb

**Comment by lvca** [comment]
Fixed in 2.0.4-hotfix and 2.1-SNAPSHOT (develop branch)

**Comment by nin-jin** [comment]
Thank you. Link above is pointer to bug place.
